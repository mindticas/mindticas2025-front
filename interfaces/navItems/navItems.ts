import { ComponentType } from 'react';

export interface NavItem {
    name?: string;
    href: string;
    icon: ComponentType<React.SVGProps<SVGSVGElement>>;
}

export interface AdminNavbarDesktopProps {
    navItems: NavItem[];
    pathname: string;
}

export interface AdminNavbarMobileProps extends AdminNavbarDesktopProps {
    isMobileMenuOpen: boolean;
    onToggleMenu: () => void;
}
