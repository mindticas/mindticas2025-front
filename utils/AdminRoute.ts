import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
