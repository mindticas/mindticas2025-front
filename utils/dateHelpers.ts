// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}