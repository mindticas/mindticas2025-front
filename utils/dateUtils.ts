import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
