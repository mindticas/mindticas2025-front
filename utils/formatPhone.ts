export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
