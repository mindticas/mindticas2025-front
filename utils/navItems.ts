import { BarChart, Calendar, Scissors, Users } from "lucide-react";

export const navItems = [
    { name: 'Citas', href: '/admin/citas', icon: Calendar },
    { name: 'Empleados', href: '/admin/empleados', icon: Users },
    { name: 'Tratamientos', href: '/admin/tratamientos', icon: Scissors },
    { name: 'Reportes', href: '/admin/reportes', icon: BarChart },
];