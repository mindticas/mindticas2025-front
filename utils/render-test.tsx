import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
