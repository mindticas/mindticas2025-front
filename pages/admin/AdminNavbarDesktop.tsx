import {
    Box,
    Flex,
    Link as ChakraLink,
    Button,
    Text,
    Image,
} from '@chakra-ui/react';
import { LogOut, Scissors } from 'lucide-react';
import React from 'react';
import NextLink from 'next/link';
import { AdminNavbarDesktopProps } from '@/interfaces/navItems/navItems';
import { handleRefresh } from '@/services/RefreshToken';
import CalendarImg from '@/public/google-calendar.png';
import { handleLogout } from '@/services/authService';

export default function DesktopViewAdmin({
    navItems,
    pathname,
}: AdminNavbarDesktopProps) {
    return (
        <Box bg='gray.800' color='white' shadow='lg'>
            <Flex
                maxW='7xl'
                mx='auto'
                px={{ base: 4, md: 6, lg: 8 }}
                h='16'
                align='center'
                justify='space-between'
            >
                {/* Logo and links */}
                <Flex align='center'>
                    <ChakraLink
                        display='flex'
                        alignItems='center'
                        _hover={{ textDecoration: 'none' }}
                        color='white'
                    >
                        <Scissors size={32} />
                        <Text ml='2' fontSize='xl' fontWeight='bold'>
                            Elegangster Admin
                        </Text>
                    </ChakraLink>
                    <Flex ml='10' align='center' gap='4'>
                        {navItems &&
                            navItems.map((item) => {
                                const Icon = item.icon;
                                const active = pathname === item.href;
                                const hasName = !!item.name;
                                return (
                                    <ChakraLink
                                        key={item.name}
                                        as={NextLink}
                                        px='3'
                                        py='2'
                                        href={item.href}
                                        rounded='md'
                                        fontSize='md'
                                        fontWeight='medium'
                                        bg={active ? 'gray.900' : 'transparent'}
                                        _hover={{
                                            bg: 'gray.700',
                                            textDecoration: 'none',
                                        }}
                                        display='flex'
                                        alignItems='center'
                                        color='white'
                                    >
                                        <Icon
                                            width={24}
                                            height={24}
                                            style={{
                                                marginRight: hasName
                                                    ? '8px'
                                                    : '0px',
                                            }}
                                        />
                                        {item.name}
                                    </ChakraLink>
                                );
                            })}
                    </Flex>
                </Flex>
                <Flex align='center' gap='4'>
                    {/* Logout button section for desktop */}
                    <Button
                        py='2'
                        px='3'
                        color='white'
                        bg='red.600'
                        _hover={{ bg: 'red.700' }}
                        fontWeight='bold'
                        display='flex'
                        alignItems='center'
                        gap='2'
                        onClick={handleLogout}
                    >
                        <LogOut size={32} />
                        Salir
                    </Button>
                    {/* Google calendar button section for desktop */}
                    <Image
                        boxSize='40px'
                        rounded='md'
                        cursor='pointer'
                        onClick={handleRefresh}
                        src={CalendarImg.src}
                        alt='Imagen de google calendar'
                    />
                </Flex>
            </Flex>
        </Box>
    );
}
