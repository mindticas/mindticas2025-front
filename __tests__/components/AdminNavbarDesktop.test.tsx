import AdminNavbarDesktop from '@/pages/admin/AdminNavbarDesktop';
import { navItems } from '@/utils/navItems';
import { render } from '@/utils/render-test';

describe('AdminNavbarDesktop Snapshot', () => {
    it('should match the snapshot', () => {
        const { asFragment } = render(
            <AdminNavbarDesktop navItems={navItems} pathname='/admin/citas' />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
