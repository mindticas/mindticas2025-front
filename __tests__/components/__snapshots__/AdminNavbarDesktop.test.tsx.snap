// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNavbarDesktop Snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-efzgkz"
  >
    <div
      class="css-lhqv6a"
    >
      <div
        class="css-70qvj9"
      >
        <a
          class="chakra-link css-1sa6w8m"
          href="/admin/citas"
        >
          <svg
            class="lucide lucide-scissors"
            fill="none"
            height="32"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="6"
              cy="6"
              r="3"
            />
            <path
              d="M8.12 8.12 12 12"
            />
            <path
              d="M20 4 8.12 15.88"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <path
              d="M14.8 14.8 20 20"
            />
          </svg>
          <p
            class="css-2jpnh5"
          >
            Elegangster Admin
          </p>
        </a>
        <div
          class="css-1xaiisq"
        >
          <a
            class="chakra-link css-v6hlzi"
            href="/admin/citas"
          >
            <svg
              class="lucide lucide-calendar"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-right: 8px;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
            Citas
          </a>
          <a
            class="chakra-link css-a3d9va"
            href="/admin/empleados"
          >
            <svg
              class="lucide lucide-users"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-right: 8px;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="9"
                cy="7"
                r="4"
              />
              <path
                d="M22 21v-2a4 4 0 0 0-3-3.87"
              />
              <path
                d="M16 3.13a4 4 0 0 1 0 7.75"
              />
            </svg>
            Empleados
          </a>
          <a
            class="chakra-link css-a3d9va"
            href="/admin/tratamientos"
          >
            <svg
              class="lucide lucide-scissors"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-right: 8px;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="6"
                cy="6"
                r="3"
              />
              <path
                d="M8.12 8.12 12 12"
              />
              <path
                d="M20 4 8.12 15.88"
              />
              <circle
                cx="6"
                cy="18"
                r="3"
              />
              <path
                d="M14.8 14.8 20 20"
              />
            </svg>
            Tratamientos
          </a>
          <a
            class="chakra-link css-a3d9va"
            href="/admin/reportes"
          >
            <svg
              class="lucide lucide-chart-no-axes-column-increasing"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-right: 8px;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="12"
                x2="12"
                y1="20"
                y2="10"
              />
              <line
                x1="18"
                x2="18"
                y1="20"
                y2="4"
              />
              <line
                x1="6"
                x2="6"
                y1="20"
                y2="16"
              />
            </svg>
            Reportes
          </a>
        </div>
      </div>
      <button
        class="chakra-button css-17n3sdn"
        type="button"
      >
        <svg
          class="lucide lucide-log-out"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
          />
          <polyline
            points="16 17 21 12 16 7"
          />
          <line
            x1="21"
            x2="9"
            y1="12"
            y2="12"
          />
        </svg>
        Salir
      </button>
    </div>
  </div>
</DocumentFragment>
`;
